### PR TITLE
IDEMPIERE-5302 - Drill Assistant Feature with re-generated migration scripts

### PR DESCRIPTION
**Pull request by PeterTakacs300**
JIRA ticket: https://idempiere.atlassian.net/browse/IDEMPIERE-5302

### CODE REVIEW

**Review by CarlosRuiz-globalqss — CHANGES_REQUESTED** [review]
Hi @PeterTakacs300,

The new migration script has some wrong sequences.  Please apply the following patch to fix the centralized IDs:
[5302_fixseq.txt](https://github.com/idempiere/idempiere/files/9084480/5302_fixseq.txt)

Also, I noticed the script is incomplete.

The new column+field AD_Table.IsShowInDrillOptions is not there.

Also, not sure if there are new indexes or foreign keys in the old script that are not present in the new, please better review those.

**Comment by PeterTakacs300** [comment]
Hi @CarlosRuiz-globalqss, I've completed the scripts with the IsShowInDrillOptions column and field, applied your patch, and after review, I've found no foreign key constraints or indexes that are needed, and missing from the new script (I've found FK constraint _celement_addrillrule_ on AD_Process_DrillRule, but it is not needed anymore)

File: migration/iD10/oracle/202207081747_IDEMPIERE-5302.sql [hunk 1]
```diff
@@ -0,0 +1,911 @@
...
+;
+
+-- Jul 12, 2022, 9:32:52 AM CEST
+INSERT INTO AD_Column (AD_Column_ID,Version,Name,Description,AD_Table_ID,ColumnName,DefaultValue,FieldLength,IsKey,IsParent,IsMandatory,IsTranslated,IsIdentifier,SeqNo,IsEncrypted,AD_Reference_ID,AD_Client_ID,AD_Org_ID,IsActive,Created,CreatedBy,Updated,UpdatedBy,AD_Element_ID,IsUpdateable,IsSelectionColumn,EntityType,IsSyncDatabase,IsAlwaysUpdateable,IsAutocomplete,IsAllowLogging,AD_Column_UU,IsAllowCopy,SeqNoSelection,IsToolbarButton,IsSecure,FKConstraintType,IsHtml) VALUES (214721,0,'Show In Drill Options','This parameter enables the table to be displayed in Drill Assistant - Table tab',100,'IsShowInDrillOptions','N',1,'N','N','N','N','N',0,'N',20,0,0,'Y',TO_TIMESTAMP('2022-07-12 09:32:52','YYYY-MM-DD HH24:MI:SS'),100,TO_TIMESTAMP('2022-07-12 09:32:52','YYYY-MM-DD HH24:MI:SS'),100,203568,'Y','N','D','N','N','N','Y','ab5fc669-6971-4f08-a7f3-160796cc2a36','Y',0,'N','N','N','N')
+;
+
+-- Jul 12, 2022, 9:32:57 AM CEST
+ALTER TABLE AD_Table ADD IsShowInDrillOptions CHAR(1) DEFAULT 'N' CHECK (IsShowInDrillOptions IN ('Y','N'))
```
Comment (CarlosRuiz-globalqss) [inline migration/iD10/oracle/202207081747_IDEMPIERE-5302.sql:16]: The default of the new column must be Y ???   For backward compatibility?

If yes, please apply the following patch:
[5302_fixdefault.txt](https://github.com/idempiere/idempiere/files/9093388/5302_fixdefault.txt)

However, I don't see a difference on this.

In my tests I generate a report from Business Partner and the Drill list is disappeared, before this patch the Drill list is pre-filled with the defaults discovered by the system.
Is this feature supposed to disappear?

Comment (PeterTakacs300) [reply migration/iD10/oracle/202207081747_IDEMPIERE-5302.sql:16]: Yes, we removed the Drill list, it was requested in the last pull request. The IsShowInDrillOptions column now only defines, which tables will appear in the Table tab of Drill Assistant. We also agreed in the previous PR, that it should be default 'N', and set 'Y' only on specific tables. There is an SQL UPDATE related to this in the migration scripts (line 28 in oracle; 25 in postgres)

Comment (CarlosRuiz-globalqss) [reply migration/iD10/oracle/202207081747_IDEMPIERE-5302.sql:16]: > Yes, we removed the Drill list, it was requested in the last pull request. The IsShowInDrillOptions column now only defines, which tables will appear in the Table tab of Drill Assistant. We also agreed in the previous PR, that it should be default 'N', and set 'Y' only on specific tables. There is an SQL UPDATE related to this in the migration scripts (line 28 in oracle; 25 in postgres)

No, I just checked the original pull request 1345 and the default of IsShowInDrillOptions is set to N (line 19 oracle) and some lines later is set to Y (line 26 oracle).

For the deprecation of the old drill functionality I'll move the discussion to the ticket.

Comment (PeterTakacs300) [reply migration/iD10/oracle/202207081747_IDEMPIERE-5302.sql]: I see, OK util then I'm fixing IsShowInDrillOptions, default to 'Y'.

**Comment by CarlosRuiz-globalqss** [comment]
Hi @PeterTakacs300 - there is a minor conflict with the last commit from IDEMPIERE-5347 (#1393)

Can you please solve the conflict?  I would help but your branch is not editable.

Regards,

Carlos Ruiz

**Comment by CarlosRuiz-globalqss** [comment]
Hi @PeterTakacs300 - the merge went bad and it's not compiling in eclipse.

Can you please check if the following patch fixes the issues:
[pr1390_badimports.txt](https://github.com/idempiere/idempiere/files/9357871/pr1390_badimports.txt)

Please never commit before checking that the project is free of errors.

**Comment by PeterTakacs300** [comment]
Hi @CarlosRuiz-globalqss, my apologies, I didn't re-build the project after merge.
Seems like your patch fixes the errors.

**Review by CarlosRuiz-globalqss — APPROVED** [review]

**Comment by CarlosRuiz-globalqss** [comment]
Hi @hengsin - for me this is OK, I did some tests and is working fine.

If it's OK for you can you please squash and merge it?

**Pull request merged**